textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcherTest.java

```diff
@@ -47,21 +49,23 @@
 public class DictionaryMatcherTest {
 
     private DictionaryMatcher dictionaryMatcher;
-    private LuceneDataStore dataStore;
+    private DataStore dataStore;
     private IDataWriter dataWriter;
     private IDataReader dataReader;
     private Analyzer analyzer;
     private Query query;
+    private IPredicate dataReaderPredicate;
 
     @Before
     public void setUp() throws Exception {
 
-        dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
+        dataStore = new DataStore(DataConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
         analyzer = new StandardAnalyzer();
-        dataWriter = new LuceneDataWriter(dataStore, analyzer);
+        dataWriter = new DataWriter(dataStore, analyzer);
         QueryParser queryParser = new QueryParser(TestConstants.ATTRIBUTES_PEOPLE[0].getFieldName(), analyzer);
-        query = queryParser.parse(LuceneConstants.SCAN_QUERY);
-        dataReader = new LuceneDataReader(dataStore, query);
+        query = queryParser.parse(DataConstants.SCAN_QUERY);
+        dataReaderPredicate = new DataReaderPredicate(dataStore, query);
```

Comment:
These two lines 67/68 show the main change caused by the refactoring compared to the old line 64.  

Old: `dataReader = new LuceneDataReader(dataStore, query);`

New:

`        dataReaderPredicate = new DataReaderPredicate(dataStore, query);
        dataReader = new DataReader(dataReaderPredicate);`

It looks good to me!